soroban-env-host/src/cost_runner/cost_types/val_deser.rs

```diff
@@ -8,5 +8,7 @@ impl CostRunner for ValDeserRun {
 
     fn run_iter(host: &crate::Host, _iter: u64, sample: Self::SampleType) {
         host.metered_from_xdr::<ScVal>(&sample).unwrap();
```

Comment:
we discussed this in chat but I'll make a note here: you might want to ensure a lot of these calls get a `black_box` wrapper as well as actually returning the constructed value to make sure it's not optimized away.

Reply:
Cool